services/worker-manager/src/api.js

```diff
@@ -217,6 +218,53 @@ builder.declare({
   return res.reply(result);
 });
 
+builder.declare({
+  method: 'post',
+  route: '/worker-pools-errors/:workerPoolId(*)',
+  name: 'reportWorkerError',
+  title: 'Report an error from a worker',
+  input: 'report-worker-error-request.yml',
+  output: 'worker-pool-error.yml',
+  scopes: {AllOf: [
+    'assume:worker-pool:<workerPoolId>',
+    'assume:worker-id:<workerGroup>/<workerId>',
+  ]},
+  stability: APIBuilder.stability.experimental,
+  description: [
+    'Report an error that occurred on a worker.  This error will be included',
+    'with the other errors in `listWorkerPoolErrors(workerPoolId)`.',
+    '',
+    'Workers can use this endpoint to report startup or configuration errors',
+    'that might be associated with the worker pool configuration and thus of',
+    'interest to a worker-pool administrator.',
+    '',
+    'NOTE: errors are publicly visible.  Ensure that none of the content',
+    'contains secrets or other sensitive information.',
+  ].join('\n'),
+}, async function(req, res) {
+  const {workerPoolId} = req.params;
+  const input = req.body;
+  const {workerGroup, workerId} = input;
+
+  await req.authorize({workerPoolId, workerGroup, workerId});
+
+  const workerPool = await this.WorkerPool.load({workerPoolId}, true);
+  if (!workerPool) {
+    return res.reportError('ResourceNotFound', 'Worker pool does not exist', {});
+  }
+
+  const wpe = await workerPool.reportError({
+    kind: input.kind,
+    title: input.title,
+    description: input.description,
```

Comment:
Since we eventually ended up adding a kludgy thing to remove fields like `password` from logs should we maybe jump the gun and do it here too before someone accidentally logs stuff?

Reply:
It's a string, though.  My hope was to just r- anything that dumps a random batch-o-data into `extra`, in preference for very specific properties.

Reply:
It's the same level of caution we exercise with `res.reportError` for example..